primitives/core/macros/src/reuse.rs

```diff
@@ -0,0 +1,404 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::ToTokens;
+use syn::{
+	parse::Parse, punctuated::Punctuated, visit_mut::VisitMut, Arm, Attribute, Block, ConstParam,
+	Error, Expr, ExprArray, ExprAssign, ExprAsync, ExprAwait, ExprBinary, ExprBlock, ExprBreak,
+	ExprCall, ExprCast, ExprClosure, ExprConst, ExprContinue, ExprField, ExprForLoop, ExprGroup,
+	ExprIf, ExprIndex, ExprInfer, ExprLet, ExprLit, ExprLoop, ExprMacro, ExprMatch, ExprMethodCall,
+	ExprParen, ExprPath, ExprRange, ExprReference, ExprRepeat, ExprReturn, ExprStruct, ExprTry,
+	ExprTryBlock, ExprTuple, ExprUnary, ExprUnsafe, ExprWhile, ExprYield, Field, FieldPat,
+	FieldValue, FnArg, GenericParam, Item, ItemConst, ItemEnum, ItemExternCrate, ItemFn,
+	ItemForeignMod, ItemImpl, ItemMacro, ItemMod, ItemStatic, ItemStruct, ItemTrait,
+	ItemTraitAlias, ItemType, ItemUnion, ItemUse, LifetimeParam, Local, Meta, Pat, PatConst,
+	PatIdent, PatLit, PatMacro, PatOr, PatParen, PatPath, PatRange, PatReference, PatRest,
+	PatSlice, PatStruct, PatTuple, PatTupleStruct, PatType, PatWild, Receiver, Result, Stmt,
+	StmtMacro, Token, TypeParam, Variant,
+};
+
+pub struct Args {
+	cases: Punctuated<Ident, Token![,]>,
+}
+
+impl Parse for Args {
+	fn parse(input: syn::parse::ParseStream) -> Result<Self> {
+		Ok(Self { cases: Punctuated::parse_terminated(input)? })
+	}
+}
+
+pub fn handle_reuse(args: Args, m: ItemMod) -> Result<TokenStream> {
+	let mut result = TokenStream::new();
+	let cases = args.cases.iter().map(ToString::to_string).collect::<Vec<_>>();
+	for current in 0..args.cases.len() {
+		let mut visitor = Visitor { cases: &cases, current, error: None };
+		let mut m = m.clone();
+		m.ident = args.cases[current].clone();
+		visitor.visit_item_mod_mut(&mut m);
+		if let Some(e) = visitor.error.take() {
```

Comment:
I have a question here, I don't see anywhere else the `error` parameter is being updated, is it safe to assume it is always going to be `None` since it was set to it when `Visitor` struct was created in `line 32`

Reply:
There's currently no error to be emitted, but the visitor may be extended in the future for futher cases, eg. attribute on raw functions rather than `mod`.
I think it necessary to always keep the ability to emit error on a clear `Span` for humanised compiling errors, rather than direct panics. Therefore I keep everywhere in proc macro returning `Result`.
Since visitor's definition does not allow returning `Result`, I use this error field as workaround, regardless of whether it would be used.

Reply:
That makes sense